libs/auth-api-lib/src/lib/delegations/delegations.service.ts

```diff
@@ -435,26 +456,49 @@ export class DelegationsService {
       // Filter out companies, since we do not need to make a national registry call for them.
       .filter(({ fromNationalId }) => !kennitala.isCompany(fromNationalId))
       .map(({ fromNationalId }) =>
-        this.nationalRegistryClient.getIndividual(fromNationalId),
+        this.nationalRegistryClient
+          .getIndividual(fromNationalId)
+          .catch(this.handlerGetIndividualError),
       )
 
     try {
       // Check if delegations is linked to a person, i.e. not deceased
       const persons = await Promise.all(delegationsPromises)
-      const personsValues = persons.filter(isDefined)
+      const personsValuesNoError = persons
+        .filter(this.isNotError)
+        .filter(isDefined)
 
       // Divide delegations into alive or deceased delegations.
-      const [aliveDelegations, deceasedDelegations] = partition(
+      const [aliveDelegations, deceasedDelegations] = partitionWithIndex(
         delegations,
-        ({ fromNationalId }) =>
+        ({ fromNationalId }, index) =>
           // All companies will be divided into aliveDelegations
           kennitala.isCompany(fromNationalId) ||
+          // Pass persons that are of Error instance, since we want do return the delegation.
+          persons[index] instanceof Error ||
           // Make sure we can match the person to the delegation, i.e. not deceased
-          personsValues.find((person) => person?.nationalId === fromNationalId),
+          personsValuesNoError.some(
+            (person) => person?.nationalId === fromNationalId,
+          ),
       )
 
+      // If Þjóðskrá throws and error for some fromNationalId, then we still want to return the delegation.
+      // however we change the delegation.fromName to unknown name.
+      const modifiedAliveDelegations = aliveDelegations
+        .map((aliveDelegation) => {
```

Comment:
Consider `aliveDelegations.forEach` to skip the extra array since we're mutating the objects anyways?

Reply:
Good point but I rather update the logic inside map to
```
      const modifiedAliveDelegations = aliveDelegations
        .map((aliveDelegation) => {
          const person = this.getPersonByNationalId(
            personsValuesNoError,
            aliveDelegation.fromNationalId,
          )

          return {
            ...aliveDelegation,
            fromName: person?.name ?? aliveDelegation.fromName ?? UNKNOWN_NAME,
          }
        })
        .filter(isDefined)
```